golang/x/relay/client/rest/query.go

```diff
@@ -289,3 +289,49 @@ func getRequestHandler(cliCtx context.CLIContext, storeName string) http.Handler
 		rest.PostProcessResponse(w, cliCtx, res)
 	}
 }
+
+// handler function for checkRequests queries. parses arguments from url string, and passes them through
+// as a QueryParamsCheckRequests struct
+func checkRequestsHandler(cliCtx context.CLIContext, storeName string) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		vars := mux.Vars(r)
+
+		var proof types.SPVProof
+		jsonErr := json.Unmarshal([]byte(vars["proof"]), &proof)
+		if jsonErr != nil {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, jsonErr.Error())
+			return
+		}
+
+		var requests []types.FilledRequestInfo
+		jsonErr = json.Unmarshal([]byte(vars["requests"]), &requests)
+		if jsonErr != nil {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, jsonErr.Error())
+			return
+		}
```

Comment:
can we combine both of these into 1 Unmarshal? maybe skip straight to  `QueryParamsCheckRequests` ?

Reply:
soooo........wondering with the best way to do this. I could take `vars` and `Marshal` that to JSON to then `Unmarshal` into `FilledRequests` but I'm still doing two marshalling calls. I could build the string from `vars` 
 to `Unmarshal`  (or build the string from parsing the http request itself, which seems a little involved)

Reply:
obviously would would be easier to toy around with the possibilities with manual testing

Reply:
or I COULD just take a json `FilledRequests` from the beginning. I split this into 2 args thinking it would make more sense on the user end, but was still unsure if it was the right decision

Reply:
okay. so I think we need to turn this into a POST request in order to have a JSON body. All the other GETs are simple URL-argument things, but this one is complex. That'll simplify this parsing